src/components/AssetTree/AssetTree.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from 'react';
+import { Tree } from 'antd';
+import { AntTreeNode } from 'antd/lib/tree';
+import {
+  AssetTreeType,
+  TreeNodeType,
+  AssetType,
+  OnSelectReturnType,
+  TreeNodeData,
+} from 'utils/validators';
+
+const { TreeNode } = Tree;
+
+class AssetTree extends Component<AssetTreeType> {
+  state = {
+    treeData: [],
+    expandedKeys: {},
+  };
+
+  async componentDidMount() {
+    const { assets, defaultExpandedKeys } = this.props;
+
+    if (assets && assets.length > 0) {
+      this.setState({
+        treeData: this.mapDataAssets(assets),
+        expandedKeys: defaultExpandedKeys
+          ? this.toKeys(defaultExpandedKeys)
+          : {},
+      });
+    }
+  }
+
+  mapDataAssets = (assets: AssetType[]) => {
+    const nodes: { [name: string]: TreeNodeData } = {};
+
+    assets.forEach(asset => {
+      nodes[asset.id] = this.returnPretty(asset);
+    });
+
+    const addedAsChildren: (number | string)[] = [];
```

Comment:
Will not this always be of type `number[]`?

Reply:
@f1cognite we push asset.id to this - which is defined as string or number - so in short - no.

if so - this is a larger change we will do. From opint id is defined as both and we've constructed our types to match :)